events/ticket_pdf_generator.py

```diff
@@ -0,0 +1,74 @@
+from reportlab.lib.pagesizes import letter
+from reportlab.graphics.barcode.qr import QrCodeWidget
+from reportlab.graphics.shapes import Drawing
+from reportlab.lib.fonts import addMapping
+from reportlab.pdfbase import pdfmetrics
+from reportlab.pdfbase.ttfonts import TTFont
+from reportlab.lib.units import inch
+from reportlab.lib.styles import getSampleStyleSheet, ParagraphStyle
+from reportlab.platypus import SimpleDocTemplate, Paragraph, Spacer, Image
+from reportlab.lib.enums import TA_JUSTIFY
+from datetime import datetime
+import io
+
+
+class TicketPdfGenerator:
+    def generate(self, participant, event):
+        buffer = io.BytesIO()
+
+        doc = SimpleDocTemplate(buffer, pagesize=letter,
+                                rightMargin=72, leftMargin=72,
+                                topMargin=72, bottomMargin=18)
+
+        payload = dict(participant_id=participant.id, event_id=event.id)
+        qrw = QrCodeWidget(payload)
+        b = qrw.getBounds()
+
+        w = b[2] - b[0]
```

Comment:
Can you add some kind of explanation what these boundaries are (indexes) so that other people could easily modify that if necessary?

Reply:
Yeah sure. I added short description in code. getBounds returns boundary as array: b[0], b[1], b[2], b[3] = x1, y1, x2, y2 of Qr code widget.